### DISCUSSION TITLE
Type and Width/Radius in Primitive

### DISCUSSION THREAD

**Discussion by aldrians** [gpuweb/gpuweb#4228] [Q&A] [answered]
I thing we miss essential properties for primitive object. 
When using topology "line-list" or "line-strip" to creating line objects we also want to describe line type (like continuous, dot, dash, or other type), the width of the line, how the line join (chamfer,fillet and it distance/radius), and base position of the line (center or outside), also in topology "point-list" to create point that have type (like dot, cross, or user define) and the radius.

**Comment by greggman** [comment, marked as answer]
WebGPU is a low-level API. If you want fancy lines you need to build your own library (or find an existing one).

See this article. It's based on WebGL but the solutions are similar for WebGPU

https://mattdesl.svbtle.com/drawing-lines-is-hard

The same is true for dots/cross/user defined points. All of that is something that would happen at a higher level than WebGPU

**Reply by aldrians** [reply]
ooh ya, the best way to implement it is in higher level as you told. thanks.
